### ISSUE TITLE
Swagger (error) response messages

### ISSUE THREAD

**Issue by salceson** [scalatra/scalatra#577]
Hi!

I'm using Scalatra Swagger to produce the documentation to my REST API and it is working very well. The only thing is missing is the error message model. I use the following snippet to declare the operation:

```scala
val searchEvidenceProfiles = (
    apiOperation[JSONSearchResponse]("searchEvidenceProfiles")
      summary "Search evidence profiles"
      (...)
      responseMessage StringResponseMessage(400, "badType")
    )
```

While this works (it creates the entry in the Swagger JSON), it does not put the error response class into Swagger JSON (see the image from Swagger UI below).

![swagger](https://cloud.githubusercontent.com/assets/3855869/16270673/69bbd2f0-3897-11e6-93d4-1113cf367cd5.png)

While in this case it is a simple class with message string, I also do have more complicated error response model that I would like to present to users.

I think that probably API that allows sth like:
```scala
responseMessageWithModel[T: Manifest: NotNothing](err: ResponseMessage[_])
```
should be also included; it would enable the developer to include the error response model.

Do you think that this is a valid point? Or maybe I'm missing something here?

Best regards,
Michał Ciołczyk

**Comment by takezoe** [comment]
@salceson Fixed in 2.5.0 branch.

**Comment by salceson** [comment]
@takezoe Thanks, great ;)
